parser/src/main/java/com/microsoft/gctoolkit/parser/GCLogTrace.java

```diff
@@ -289,4 +302,13 @@ public void notYetImplemented() {
         System.out.println("-----------------------------------------");
         //}
     }
+
+    public ZGCPhase getZCollectionPhase() {
```

Comment:
A GCLogTrace is a convenience wrapper over the Matcher. It contains convenience methods to extract data from the matcher. It is expected that the parser has the context to know which methods to call and which groups need to be extracted. Thus ZGCPhase is either known and extracted by the parser at that time *or* ZGCPhase isn't present in the log line (matcher) and the parser, again, should know this and not call this method. Either way, ZGCPhase should be consistent and there should be no need for downstream exception handling. This is something that was noted in ZGCPhase that an illegal argument was being thrown which likely is unnecessary.